spotlight-client/src/contentModels/Metric.ts

```diff
@@ -119,13 +132,15 @@ export default class Metric<RecordFormat extends AnyRecord> {
       metricNames: [this.sourceFileName],
       tenantId: this.tenantId,
     });
-    if (apiResponse) {
-      const metricFileData = apiResponse[this.sourceFileName];
-      if (metricFileData) {
-        this.allRecords = this.dataTransformer(metricFileData);
+    runInAction(() => {
+      if (apiResponse) {
+        const metricFileData = apiResponse[this.sourceFileName];
+        if (metricFileData) {
+          this.allRecords = this.dataTransformer(metricFileData);
+        }
+        this.isLoading = false;
```

Comment:
Is your intention to eventually store the metric data as part of the `TenantStore` using `getMetricsForTenant`?

Reply:
I did it this way because the relationship between tenant and metric is strictly hierarchical ... the [Mobx docs](https://mobx.js.org/defining-data-stores.html#domain-objects) advise against normalizing those kinds of relationships or making your stores resemble relational database tables or something like that. Instead the metrics are literally nested within `TenantStore.currentTenant`.

As to your second question, it's a good question ... I don't really know the answer yet. There are some places where a `currentMetric` would make sense (e.g. you have navigated to the page for a single metric), but in other places you will see multiple metrics at once. There is nothing nested below a `Metric` so it wouldn't serve the same limiting function to designate one as "current" and thus it might not even be necessary to do it at all? Figuring this out will be a big part of #274.